server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -2318,6 +2319,11 @@ public ServiceOffering createServiceOffering(final CreateServiceOfferingCmd cmd)
             }
         }
 
+        // check if valid cache_mode parameter
+        if(isCacheModeInvalid(cmd.getCacheMode())){
+            throw new InvalidParameterValueException("Please specify a valid cache mode parameter");
+        }
+
```

Comment:
I think it is better to call the method `validateCachMode(String cacheMode)` and include this code as it is duplicated below in lines 2951 - 2955:
```java
void validateCachMode(String cacheMode){
    if( cacheMode != null &&
        !Enums.getIfPresent(DiskOffering.DiskCacheMode.class,
                            cacheMode.toUpperCase()).isPresent() {
        throw new InvalidParameterValueException("Please specify a valid cache mode parameter");
    }
}
```